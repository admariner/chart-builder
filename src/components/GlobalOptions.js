// @flow
import React, { Component } from 'react'
import { FormControl, FormGroup, InputGroup, HelpBlock } from 'react-bootstrap'
import { observer, inject } from 'mobx-react'
import classes from './GlobalOptions.module.css'

import type { StoreType } from '../util/Store'

type Props = {
  store: StoreType
}

class GlobalOptions extends Component<Props> {
  render() {
    const { store } = this.props
    return (
      <div>
        <div className={classes.dimensions}>
          <FormGroup bsSize="sm" className={classes.formGroup}>
            <InputGroup>
              <FormControl
                type="number"
                placeholder="Auto"
                value={store.config.width || ''}
                onChange={e => {
                  const val = parseInt(e.target.value)
                  if (!isNaN(val)) {
                    store.config.setDimensions(val, store.config.height)
                  }
                }}
              />
              <HelpBlock>Width</HelpBlock>
            </InputGroup>
          </FormGroup>
          <div className={classes.cross}>Ã—</div>
          <FormGroup bsSize="sm" className={classes.formGroup}>
            <InputGroup>
              <FormControl
                type="number"
                placeholder="Auto"
                value={store.config.height || ''}
                onChange={e => {
                  const val = parseInt(e.target.value)
                  if (!isNaN(val)) {
                    store.config.setDimensions(store.config.width, val)
                  }
                }}
              />
              <HelpBlock>Height</HelpBlock>
            </InputGroup>
          </FormGroup>
        </div>
      </div>
    )
  }
}

export default inject('store')(observer(GlobalOptions))
