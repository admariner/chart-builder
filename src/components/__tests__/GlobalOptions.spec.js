import React from 'react'
import Store from '../../util/Store'
import GlobalOptions from '../GlobalOptions'

it('renders', () => {
  const store = Store.create({
    config: {
      width: 100,
      height: 100
    }
  })

  snap(<GlobalOptions store={store} />)
})
