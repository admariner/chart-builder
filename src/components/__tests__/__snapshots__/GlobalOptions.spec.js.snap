// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div
    className="dimensions"
  >
    <div
      className="formGroup form-group form-group-sm"
    >
      <span
        className="input-group"
      >
        <input
          className="form-control"
          id={undefined}
          onChange={[Function]}
          placeholder="Auto"
          type="number"
          value={100}
        />
        <span
          className="help-block"
        >
          Width
        </span>
      </span>
    </div>
    <div
      className="cross"
    >
      Ã—
    </div>
    <div
      className="formGroup form-group form-group-sm"
    >
      <span
        className="input-group"
      >
        <input
          className="form-control"
          id={undefined}
          onChange={[Function]}
          placeholder="Auto"
          type="number"
          value={100}
        />
        <span
          className="help-block"
        >
          Height
        </span>
      </span>
    </div>
  </div>
</div>
`;
